include/godot_cpp/classes/ref.hpp

```diff
@@ -246,7 +268,7 @@ struct PtrToArg<Ref<T>> {
 	typedef Ref<T> EncodeT;
 
 	_FORCE_INLINE_ static void encode(Ref<T> p_val, const void *p_ptr) {
-		*(void **)p_ptr = p_val->_owner;
+		godot::internal::gdn_interface->reference_assign((GDNativeObjectPtr)p_ptr, (GDNativeObjectPtr)p_val->_owner);
```

Comment:
Requires https://github.com/godotengine/godot/pull/55277, same issue as @kidrigger is working on: https://github.com/godotengine/godot-cpp/pull/660

Reply:
removed this here for now seeing we're still discussing if we even want this change.